packages/nx-aws-cache/src/tasks-runner/aws-cache.ts

```diff
@@ -170,29 +178,42 @@ export class AwsCache implements RemoteCache {
     }
   }
 
-  private async uploadFile(hash: string, tgzFilePath: string): Promise<void> {
-    const tgzFileName = this.getTgzFileName(hash);
-    const params: clientS3.PutObjectCommand = new clientS3.PutObjectCommand({
-      Bucket: this.bucket,
-      Key: this.getS3Key(tgzFileName),
-      Body: createReadStream(tgzFilePath),
-    });
+  private getS3Key(tgzFileName: string) {
+    return join(this.path, tgzFileName);
+  }
 
+  /**
+   * When upload file with transform stream you don't know what will be the final ContentLength,
```

Comment:
Use neutral form: 
When uploading a file with a transform stream, the final ContentLength is unknown so it has to be uploaded as multipart.